rustc_codegen_spirv/src/builder_spirv.rs

```diff
@@ -71,17 +71,24 @@ pub struct BuilderSpirv {
 }
 
 impl BuilderSpirv {
-    pub fn new(version: Option<(u8, u8)>, kernel_mode: bool) -> Self {
+    pub fn new(
+        version: Option<(u8, u8)>,
+        memory_model: Option<MemoryModel>,
+        kernel_mode: bool,
+    ) -> Self {
         let mut builder = Builder::new();
         // Default to spir-v 1.3
         let version = version.unwrap_or((1, 3));
         builder.set_version(version.0, version.1);
+        let memory_model = memory_model.unwrap_or(MemoryModel::Vulkan);
         if kernel_mode {
             builder.capability(Capability::Kernel);
         } else {
             builder.extension("SPV_KHR_vulkan_memory_model");
```

Comment:
is this extension actually needed?

Reply:
Yes, the vulkan memory model isn't present for spir-v versions less than 1.5. I'll fix this, though.